prow/tide/tide.go

```diff
@@ -822,11 +822,19 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
 	for i, pr := range prs {
 		log := log.WithFields(pr.logFields())
 		mergeMethod := c.config().Tide.MergeMethod(sp.org, sp.repo)
-		if squashLabel := c.config().Tide.SquashLabel; squashLabel != "" {
+		squashLabel := c.config().Tide.SquashLabel
```

Comment:
Please add unit tests for this logic to tide_test.go

Reply:
Should this be part of the existing `TestTakeAction`? I don't see any existing tests around the previous behavior for `SquashLabel` but agree it would be useful! I could just use a little help getting started on them please :)

Reply:
a) since I'm laze I'd be inclined to encapsulate this function into its own method, aka something like:


```golang
func mergeMethod(labels []github.Label, squash, rebase, merge, method string) (string, error) {
  if squash == rebase == merge == "" { return method, nil }
  var found bool
  for _, label := range labels {
    if label not in [squash, rebase, merge] { continue }
    if found { return "", errors.New("duplicates") }
    method = label
    found = true
  }
  return method, nil
}
```

which then makes things easy to test

```golang
func TestMergeMethod(t *testing.T) {
  cases := []struct{
    name string
    labels []github.Label
    default string
    expected string
    err bool
  }{
    {
      name: "use default when no labels",
      default: github.SquashMerge,  // probably should validate two defaults
      expected: github.SquashMerge,
    },
    { name: "ignore irrelevant labels", }, //details tbd
    {
      name: "disallow multiple matches",
      labels: []github.Label{squash, rebase},
      err: true,
    },
    {
      name: "label overrides default",
      labels: []github.Label{rebase},
      default: SquashMerge,
      expected: Rebase,
    }
}

  for _, tc := range cases {
    t.Run(tc.name, func(t *testing.T) {
      actual, err := mergeMethod(tc.labels, squashLabel, rebaseLabe, mergeLabel, tc.default)
      if err != nil && !tc.err { t.Errorf("unexpected error: %v", err) }
      if actual != expected { t.Errorf("%q != expected %q", actual, expected)
  }
}
```

Reply:
cool that makes sense, thanks! will write up that change shortly.